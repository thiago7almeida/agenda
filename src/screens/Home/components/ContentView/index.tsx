import React, {useMemo} from 'react';
import {List, ListItem, useStyleSheet} from '@ui-kitten/components';
import {useSelector} from 'react-redux';

import SearchHeader from '../SearchHeader';
import RigthIcon from '../RigthIcon';
import LeftIcon from '../LeftIcon';
import themedStyles from './themedStyles';

import {ApplicationState} from '../../../../store/';
import {Contact} from '../../../../store/ducks/contacts';
import {useNavigation} from '@react-navigation/native';

//remove accents and capital letters
const normalizeText = (text: string): string =>
  text
    .toLowerCase()
    .normalize('NFD')
    .replace(/([\u0300-\u036f]|[^0-9a-zA-Z])/g, '');

const ContentView: React.FC = (): React.ReactElement => {
  const navigation = useNavigation();
  const styles = useStyleSheet(themedStyles);

  const {data} = useSelector((state: ApplicationState) => state.contacts);
  const {name} = useSelector((state: ApplicationState) => state.search);

  const filteredData: Contact[] = useMemo(() => {
    if (name.length === 0) return data;
    return data.filter((item) =>
      normalizeText(item.name).includes(normalizeText(name)),
    );
  }, [data, name]);

  return (
    <List
      data={filteredData}
      contentContainerStyle={styles.contentContainerStyle}
      renderItem={(props) => (
        <ListItem
          {...props}
          onPress={() =>
            navigation.navigate('Details', {
              isNew: false,
            })
          }
          style={styles.listItem}
          title={props.item.name}
          description={props.item.phone}
          accessoryLeft={() => LeftIcon(props.item.name)}
          accessoryRight={RigthIcon}
        />
      )}
      ListHeaderComponent={SearchHeader}
    />
  );
};

export default ContentView;
