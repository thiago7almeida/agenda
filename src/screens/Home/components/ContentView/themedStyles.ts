import {StyleService} from '@ui-kitten/components';

export default StyleService.create({
  listItem: {marginHorizontal: 10},
  contentContainerStyle: {backgroundColor: 'background-basic-color-1'},
  containerStyle: {backgroundColor: 'background-basic-color-1'},
});
