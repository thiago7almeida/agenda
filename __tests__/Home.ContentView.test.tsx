import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ContentView from '../src/screens/Home/components/ContentView';
import * as redux from 'react-redux';

const spy = jest.spyOn(redux, 'useSelector');

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useNavigation: () => {
      navigation: () => jest.fn();
    },
  };
});

describe('ContentView', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      spy.mockReturnValue({data: [], name: ''});
      const component = shallow(<ContentView />);
      expect(toJson(component)).toMatchSnapshot(
        'Primary ContentView  snapshot',
      );
    });
    it('should match to snapshot - Second', () => {
      spy.mockReturnValue({data: [{name: 'nome'}], name: ''});
      const component = shallow(<ContentView />);
      expect(toJson(component)).toMatchSnapshot('Second ContentView snapshot');
    });
    it('should match to snapshot - Third', () => {
      spy.mockReturnValue({data: [{name: 'nome'}], name: 'n'});
      const component = shallow(<ContentView />);
      expect(toJson(component)).toMatchSnapshot('Third ContentView snapshot');
    });
    it('should match to snapshot - Fourth', () => {
      spy.mockReturnValue({data: [{name: 'nome'}], name: 'd'});
      const component = shallow(<ContentView />);
      expect(toJson(component)).toMatchSnapshot('Fourth ContentView snapshot');
    });
  });
});
