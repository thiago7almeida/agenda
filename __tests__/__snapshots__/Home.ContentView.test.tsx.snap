// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentView Rendering should match to snapshot - Fourth: Fourth ContentView snapshot 1`] = `
<List
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
  data={Array []}
  renderItem={[Function]}
  style={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
/>
`;

exports[`ContentView Rendering should match to snapshot - Primary: Primary ContentView  snapshot 1`] = `
<List
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
  data={Array []}
  renderItem={[Function]}
  style={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
/>
`;

exports[`ContentView Rendering should match to snapshot - Second: Second ContentView snapshot 1`] = `
<List
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
  data={
    Array [
      Object {
        "name": "nome",
      },
    ]
  }
  renderItem={[Function]}
  style={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
/>
`;

exports[`ContentView Rendering should match to snapshot - Third: Third ContentView snapshot 1`] = `
<List
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
  data={
    Array [
      Object {
        "name": "nome",
      },
    ]
  }
  renderItem={[Function]}
  style={
    Object {
      "backgroundColor": "background-basic-color-1",
    }
  }
/>
`;
